src/GitHub.App/Models/RepositoryHosts.cs

```diff
@@ -157,11 +157,15 @@ public IRepositoryHost LookupHost([AllowNull] HostAddress address)
                     );
                     if (successful)
                     {
+                        // We need to add the connection before assigning to GitHubHost or
+                        // EnterpriseHost so that anything listening to those will find the
+                        // associated connection.
+                        connectionManager.AddConnection(address, usernameOrEmail);
```

Comment:
Well as far as I could see, getting the notification from the connection manager requires listening to `Connections.CollectionChanged` and nothing I could see relied on the `GitHubHost`/`EnterpriseHost` fields being set there. I may have missed something though.